pkg/filter/filter.go

```diff
@@ -69,13 +75,22 @@ func (f VulnerabilityFilter) FilterVulnerabilities(ctx context.Context, asset do
 	filteredVulnerabilities := make([]domain.Vulnerability, 0)
 	for _, vuln := range vulnerabilities {
 		switch {
+		case vuln.Status == invulnerable || vuln.Status == noResults:
+			logger.Info(logs.VulnerabilityFiltered{
+				Action:  logs.VulnDiscarded,
+				VulnID:  vuln.ID,
+				AssetID: asset.ID,
+				Status:  vuln.Status,
+			})
+			stater.Count("event.nexposevulnerability.filter.discarded", 1, fmt.Sprintf("reason:%s", "status"))
```

Comment:
Suggestion: I'd split this into two cases, and change "status" to "invulnerable" or "no-results", making use of the constants.

Reply:
fixed